File: validation/validation-annotation-processor/src/test/java/ru/tinkoff/kora/validation/annotation/processor/testdata/Baby.java
package ru.tinkoff.kora.validation.annotation.processor.testdata;

import ru.tinkoff.kora.validation.annotation.NotEmpty;
import ru.tinkoff.kora.validation.annotation.Pattern;
import ru.tinkoff.kora.validation.annotation.Size;
import ru.tinkoff.kora.validation.annotation.Validated;

import javax.annotation.Nullable;
import java.time.OffsetDateTime;

@Validated
public record Baby(@NotEmpty @Pattern("\\d+") String number,
                   @Size(from = 1, to = 10) Long code,
                   @Nullable OffsetDateTime timestamp,
                   Yoda yoda) {
}

File: validation/validation-symbol-processor/src/test/kotlin/ru/tinkoff/kora/validation/symbol/processor/testdata/Baby.kt
package ru.tinkoff.kora.validation.symbol.processor.testdata

import ru.tinkoff.kora.validation.annotation.*
import java.time.OffsetDateTime
import javax.annotation.Nullable

@Validated
data class Baby(
    @NotEmpty @Pattern("\\d+")
    val number: String,
    @Range(from = 1.0, to = 10.0)
    val code: Long,
    val timestamp: OffsetDateTime,
    @Validated
    val yoda: Yoda?
)
